### DISCUSSION TITLE
Slow Network Speeds in Browser

### DISCUSSION THREAD

**Discussion by DaleJP** [nextcloud/all-in-one#580] [Questions] [answered]
I am really new to this, please explain it like I'm a 9yr old:

I moved from a regular install in Ubuntu to using the AIO installer in an Ubuntu VM inside TrueNAS Scale.

First of all, everything works FANTASTICALLY. I am over the moon with how simple and fast it was to set up, BUT file transfer speeds are at a crawl. They jump between about 60KB/s to about 1.5 MB/s and I cannot play videos within the GUI. The videos take too long to buffer and it is awful.

The Ubuntu VM that the container runs in gets a Speedtest.net speed of 400Mbps both ways and the VM has a NIC attached. Cloudflare is acting as my DNS which is the only variable I can think of. Everything is pretty much out of the box. I've themed it and added the users but that is about it.

I would really appreciate some advice on how to troubleshoot and hopefully solve this problem. Everything else is working flawlessly and I really don't want to mess it up.

UPDATE:

Ok, this is really strange. I just rebooted the VM for the 20th time and it is working as normal.

I'd love to know what I should do in the future, though. There wasn't anything in the GUI logs and I'm pretty lost in the docker file structure.

The VM has 64GB of RAM, a pretty fast and large ZFS volume, and 8c 16t attached. it barely seems to utilise what it has and I wonder if there are any options I can tweak in this container to make full use of the hardware? I use it for a small school and the non-techy teachers need to use it without getting too frustrated. I need it to be as snappy and reliable as possible.

I know this is generally a place for those that are already very knowledgeable so please forgive my muddying the water.

**Comment by szaimen** [comment, marked as answer]
Hello, thanks for the positive feedback and there was this thread already: https://github.com/nextcloud/all-in-one/discussions/538

**Reply by DaleJP** [reply]
Hello!

Brilliant, thank you. I'll look into PiHole and check my firewall.

**Reply by DaleJP** [reply]
Sorry, while I have you here, is there anything you can point me to to allow NC the full use of the system and improve performance? I've seen some "server tweaking" posts from a long time ago, but the file structure is very different and I'm not sure how to edit PHP settings and the like in this instance.

I'd love to have all our students/parents on the server as well as my staff and I want it performing really nicely before I do that.

Only if you have the time and inclination.

I sorted out my firewall and everything seems to be fine so far BTW. Thanks for pointing me in the right direction.
